File: src/main/java/com/musinsa/subject/repository/ProductRepositoryCustom.java
package com.musinsa.subject.repository;

import com.musinsa.subject.entity.Product;

import java.util.List;

public interface ProductRepositoryCustom {

    List<Product> getLowestPriceProductByCategory();

    List<Product> getLowestPriceProductByBrandAndCategory();

    List<Product> getMinPriceProductInCategory(long categoryId);

    List<Product> getMaxPriceProductInCategory(long categoryId);

}

File: src/main/kotlin/com/musinsa/subject/repository/ProductRepositoryCustom.kt
package com.musinsa.subject.repository

import com.musinsa.subject.entity.Product

interface ProductRepositoryCustom {
    fun getLowestPriceProductByCategory(): List<Product>
    fun getLowestPriceProductByBrandAndCategory(): List<Product>
    fun getMinPriceProductInCategory(categoryId: Long): List<Product>
    fun getMaxPriceProductInCategory(categoryId: Long): List<Product>
}
